.github/workflows/release.yml

```diff
@@ -1,28 +1,33 @@
 ---
-name: release
+name: Release
 
 permissions:
   contents: read
 
 on:
   workflow_dispatch:
     inputs:
-      branch_specifier:
-        description: The branch to release ex. main or 0.6.
+      ref:
+        description: 'Branch or tag ref to run the workflow on'
         required: true
         default: "main"
-        type: string
-
+      version:
+        description: 'The version to release (e.g. 1.2.3). This workflow will automatically perform the required version bumps'
+        required: true
```

Comment:
This is because by default the version in the pom remains `1.5.1-SNAPSHOT` after executing the `1.5.0` release, and until the changes are merged in `main` we don't know in advance if the next release will be a minor or a patch release.

As an alternative we could bump the release version when merging the first non-bugfix change that makes it a non-patch release, but I'm pretty sure that we will always forget about it.

Reply:
I think that asking the version to release upfront allows to properly name the PR and the branch name, hence making it a bit easier to work with.

If we want to change the release version, then closing the prepare PR and start over should be simpler for us. Apart from manually fixing the changelog there is little we would have to do manually in this PR (and due to the branch protection rules that would at least require 2 people to then approve the PR).

If you are worried about any potential shell injection here, then maybe we should properly validate the input value through a regex to ensure it's a `major.minor.patch` version.

Reply:
> If you are worried about any potential shell injection here, then maybe we should properly validate the input value through a regex to ensure it's a major.minor.patch version.

Yep, in addition to validating if the version has already been released to avoid the hypothetical case of a typo in the version.

Reply:
Ref: https://github.com/elastic/ecs-logging-java/pull/226/commits/0579f47cfdc7c7b0b0a716ea81bf4032deb43d96